rbd/snapshot_namespace.go

```diff
@@ -37,3 +42,35 @@ func (image *Image) GetSnapNamespaceType(snapID uint64, namespace *SnapNamespace
 		(*C.rbd_snap_namespace_type_t)(namespace))
 	return getError(ret)
 }
+
+// GetSnapNamespaceTrash returns the original name of the snapshot which was
+// moved to Trash. Caller should make sure that the snapshot ID passed in this
+// function belongs to a snapshot already in Trash.
+//
+// Implements:
+//  int rbd_snap_get_trash_namespace(rbd_image_t image, uint64_t snap_id, char *original_name, size_t max_length)
+func (image *Image) GetSnapNamespaceTrash(snapID uint64) (string, error) {
```

Comment:
The naming here is a bit odd. I'm good with moving the "get" verb to the first term in the name, I don't know why "Namespace" and "Trash" are swapped. I think "GetSnapTrashNamespace" is better.

Reply:
This as well